### PR TITLE
Refactor hook to use the github event server

### PR DESCRIPTION
**Pull request by droslean**
The github event server was created following the hook's concept. So currently hook is using so much duplicated code. 
This PR removes all the standalone event server functionality and use the github event server instead.

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by droslean** [comment]
/hold

I am holding it to do some manually end-to-end testing

File: prow/cmd/hook/main.go [hunk 1]
```diff
@@ -223,42 +224,43 @@ func main() {
...
+		ClientAgent: clientAgent,
+		ConfigAgent: configAgent,
+		Plugins:     pluginAgent,
+		Metrics:     promMetrics,
+		RepoEnabled: o.githubEnablement.EnablementChecker(),
 	}
+
+	eventServer := githubeventserver.New(o.githubEventServerOptions, secretAgent.GetTokenGenerator(o.webhookSecretFile), logrus.NewEntry(logrus.New()))
+
+	eventServer.RegisterReviewEventHandler(server.HandleReviewEvent)
```
Comment (stevekuznetsov) [inline prow/cmd/hook/main.go:243]: The event server should just handle multiple plugins for one event, not needing the second-level server to do the multiplexing.

Comment (droslean) [reply prow/cmd/hook/main.go:243]: This functionality is registered via the external plugins in [here](https://github.com/kubernetes/test-infra/pull/20717/files#diff-25f3adc74c147e59649ac9242dc3b9f462de7682b53810df1f8afe06485004fdR252)

**Pull request closed without merging**